### PR TITLE
Jetpack Cloud: Add Monitor product lightbox, product data

### PR DESCRIPTION
**Pull request by mattgawarecki**
***UPDATE:** D117122-code is deployed. All relevant instructions have been struck through.*

~~**NOTE: This PR is not fit to merge until a product has been created for Jetpack Monitor in the WordPress.com Store.**~~

~~Depends on D117122-code.~~
Resolves `1204992567518369-as-1204995327932145`.

## Proposed Changes

* Add a section for disclaimer text to `JetpackProductInfo`.
* Add a property `alwaysExpanded` to `JetpackProductInfoSection`, to force a section to always render in its expanded state.
* Add product and feature slugs for Jetpack Monitor.
* Add lightbox verbiage and icons for Jetpack Monitor.

## Testing Instructions

~~***NOTE:** To test before D117122-code is merged, you'll need a WordPress.com sandbox. For prerequisite test instructions, consult the aforementioned PR.*~~

1. In your test environment, visit the Partner Portal's **Issue License** page (i.e., `/partner-portal/issue-license`).
2. Verify you see Monitor in the list of issuable products.
3. Click the "More about Monitor" link to open its lightbox.
4. Verify the lightbox opens, and that all its verbiage and icons are rendered the same as in its design (Gk3oZYJHerhUZ2KVabMAEx-fi-4016-92602).
5. Click the **Select License** button; verify the lightbox closes and the product card's checkbox is now checked.

### CODE REVIEW

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8607527

Thank you @mattgawarecki for including a screenshot in the description! This is really helpful for our translators.

File: packages/calypso-products/src/products-list.ts [hunk 1]
```diff
@@ -432,6 +440,40 @@ export const JETPACK_SITE_PRODUCTS_WITH_FEATURES: Record<
 		getProductId: () => 2400,
 		getStoreSlug: () => PRODUCT_JETPACK_BOOST_MONTHLY,
 	},
+
+	[ PRODUCT_JETPACK_MONITOR_YEARLY ]: {
+		product_name: translate( 'Monitor' ),
```
Comment (yashwin) [inline packages/calypso-products/src/products-list.ts:457]: As per the design, the title is `Downtime Monitoring.` I might be missing something here.

**Review by yashwin — COMMENTED** [review]
I reviewed this since it was marked as `Ready for review`.

This works as expected. Great job on this! 

A few clarifications and issues I found

- The product name in this PR is `Monitor`, whereas it is `Downtime Monitoring` in the design.
- There is a slight alignment issue in the disclaimer section. Is this intentional? 

Screenshot 2023-08-01 at 11 51 42 AM


- The ticket also mentions the modal should open up from the JITM and upgrade paths from the dashboard. Are we planning to wrap that in a separate PR? If yes, could we please create different ticket so that we don't miss it.

**Comment by mattgawarecki** [comment]
> The product name in this PR is `Monitor`, whereas it is `Downtime Monitoring` in the design.

Ah, yes -- this is intentional, because of this discussion: pedEI5-NG-p2#comment-498. I suspect the designs might've pre-dated this decision.

> There is a slight alignment issue in the disclaimer section. Is this intentional?

Fixed 👍 

> The ticket also mentions the modal should open up from the JITM and upgrade paths from the dashboard. Are we planning to wrap that in a separate PR? If yes, could we please create different ticket so that we don't miss it.

Good catch! I added event handlers to those spots. 👍

**Comment by yashwin** [comment]
@mattgawarecki: Thanks for addressing my comments!

I see a couple more upgrade paths where the modal should be opened. 

Enable monitor if not enabled already > Click on the clock icon next to the monitor toggle.

- Hover over the `Upgrade` badge > Click `Explore`

Screenshot 2023-08-03 at 11 02 20 AM

- Click the `Upgrade ($0.91/m)` link

Screenshot 2023-08-03 at 11 02 37 AM

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8607527

Thank you @mattgawarecki for including a screenshot in the description! This is really helpful for our translators.

**Comment by mattgawarecki** [comment]
@yashwin Nice -- I made changes to address those two cases. Test one more time for me?

**Review by yashwin — APPROVED** [review]
LGTM 🚀 Thanks for making the changes. Good job on this!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8607527).

**Pull request merged**